internal/retry/retry.go

```diff
@@ -79,3 +79,19 @@ type Doer interface {
 	// It should return an error that can be checked for retriability.
 	Do(ctx context.Context) error
 }
+
+// DoerFunc is a convenience implementation of Doer for a closure.
+type DoerFunc func(context.Context) error
+
+// Do implements Doer.
+func (f DoerFunc) Do(ctx context.Context) error {
+	return f(ctx)
+}
+
+// Ensure DoerFunc implements Doer.
+var _ = Doer(DoerFunc(func(context.Context) error { return nil }))
+
+// Always can be used as an argument to NewIntervalRetrier to always retry.
+func Always(_ error) bool {
+	return true
+}
```

Comment:
```suggestion
func Always(error) bool { return true }
```

nit: Align this with the inlining of the function above.